apps/studio/src/pages/dashboard/edit/index.tsx

```diff
@@ -1,26 +1,28 @@
-import { useEffect } from "react"
-import { Grid, GridItem } from "@chakra-ui/react"
-
-import { useEditorDrawerContext } from "~/contexts/EditorDrawerContext"
-import EditPageDrawer from "~/features/editing-experience/components/EditPageDrawer"
-import Preview from "~/features/editing-experience/components/Preview"
-import { PageEditingLayout } from "~/templates/layouts/PageEditingLayout"
-import { trpc } from "~/utils/trpc"
+import { Grid, GridItem } from '@chakra-ui/react'
+import { useEffect } from 'react'
+import { useEditorDrawerContext } from '~/contexts/EditorDrawerContext'
+import EditPageDrawer from '~/features/editing-experience/components/EditPageDrawer'
+import Preview from '~/features/editing-experience/components/Preview'
+import { PageEditingLayout } from '~/templates/layouts/PageEditingLayout'
+import { trpc } from '~/utils/trpc'
 
 function EditPage(): JSX.Element {
-  const { setDrawerState, pageState, setPageState, setEditorState } =
-    useEditorDrawerContext()
+  const {
+    setDrawerState,
+    pageState,
+    setPageState,
+    setSnapshot: setEditorState,
+  } = useEditorDrawerContext()
```

Comment:
feels weird that this page itself does not have its own context and has to use a context further up. Any reason why the context is not inside this component and is instead outside? Seems like all the props are initialised in this component.

Reply:
sounds good, lemme just shift it in here

Reply:
i think this got changed somewhere :monkas: